identity-wallet/src/state/connections/mod.rs

```diff
@@ -1,30 +1,182 @@
 pub mod actions;
 pub mod reducers;
 
-use super::FeatTrait;
+use super::{core_utils::DateUtils, FeatTrait};
 
+use log::info;
 use serde::{Deserialize, Serialize};
+use std::ops::Not;
 use ts_rs::TS;
 
-/// Connection contains the ID and information of a connection.
-#[derive(Clone, Serialize, Debug, Deserialize, TS, PartialEq, Default)]
+#[derive(Default, Serialize, Deserialize, Clone, Debug, TS, PartialEq)]
+#[ts(export)]
+pub struct Connections(pub Vec<Connection>);
+
+impl Connections {
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn contains(&self, url: &str, name: &str) -> bool {
+        self.0
+            .iter()
+            .any(|connection| connection.url == url && connection.name == name)
+    }
+
+    /// Inserts a new connection into the list of connections.
+    /// Modelled after the `std::collections::HashMap::insert` method.
+    fn insert(&mut self, connection: Connection) -> Option<&Connection> {
+        self.contains(&connection.url, &connection.name)
+            .not()
+            .then(|| {
+                self.0.push(connection);
+                self.0.last()
+            })
+            .flatten()
+    }
+
+    /// Returns a mutable reference to the connection with the given `url` and `name`.
+    /// Modelled after the `std::collections::HashMap::get_mut` method.
+    fn get_mut(&mut self, url: &str, name: &str) -> Option<&mut Connection> {
+        self.0
+            .iter_mut()
+            .find(|connection| connection.url == url && connection.name == name)
+    }
+
+    /// Inserts a new connection into the list of connections if it does not already exist. If it does exist, updates
+    /// the last interaction time and returns a reference to the connection.
+    pub fn insert_or_update(&mut self, url: &str, name: &str) -> &Connection {
```

Comment:
Since we're first checking if it exists and _then_ inserting if it doesn't, I would rename to `update_or_insert`.

Reply:
good point :sweat_smile: